benchmarks/benchmark_db_utils.py

```diff
@@ -0,0 +1,277 @@
+  ) -> bq_writer_utils.create_bq_writer_object:
+    """Creates a BigQuery client object.
+
+    Args:
+      table: The name of the BigQuery table.
+      dataclass_type: The dataclass type corresponding to the table schema.
+      is_test: Whether to use the testing project or the production project.
+
+    Returns:
+      A BigQuery client object.
+    """
+
+    project = "supercomputer-testing" if is_test else "ml-workload-benchmarks"
+    dataset = "mantaray_v2" if is_test else "benchmark_dataset_v2"
+    return bq_writer_utils.create_bq_writer_object(
+        project=project,
+        dataset=dataset,
+        table=table,
+        dataclass_type=dataclass_type,
+    )
+
+  def _validate_id(
+      id_value: str,
+      table_name: str,
+      id_field: str,
+      dataclass_type: Type,
+      is_test: bool = False,
+  ) -> bool:
+    """Generic function to validate an ID against a BigQuery table.
+
+    Args:
+      id_value: The ID value to validate.
+      table_name: The name of the BigQuery table.
+      id_field: The name of the ID field in the table.
+      is_test: Whether to use the testing project or the production project.
+
+    Returns:
+      True if the ID is valid, False otherwise.
+    """
+
+    client = get_db_client(table_name, dataclass_type, is_test)
+    result = client.query(where={id_field: id_value})
+
+    if not result:
+      logger.info(
+          "%s: %s is not present in the %s table ",
+          id_field.capitalize(),
+          id_value,
+          table_name,
+      )
+      logger.info(
+          "Please add %s specific row in %s table before adding to run summary table",
+          id_value,
+          table_name,
+      )
+      return False
+    return True
+
+  def validate_model_id(model_id: str, is_test: bool = False) -> bool:
+    """Validates a model ID against the model_info table."""
+
+    print("model id: " + model_id)
+    id_val = _validate_id(
+        model_id, "model_info", "model_id", model_info_schema.ModelInfo, is_test
+    )
+    if not id_val:
+      print("model id validation failed")
+      return False
+    return True
+
+  def validate_hardware_id(hardware_id: str, is_test: bool = False) -> bool:
+    """Validates a hardware ID against the hardware_info table."""
+    id_val = _validate_id(
+        hardware_id,
+        "hardware_info",
+        "hardware_id",
+        hardware_info_schema.HardwareInfo,
+        is_test,
+    )
+    if not id_val:
+      print("hardware id validation failed")
+      return False
+    return True
+
+  def validate_software_id(software_id: str, is_test: bool = False) -> bool:
+    """Validates a software ID against the software_info table."""
+    id_val = _validate_id(
+        software_id,
+        "software_info",
+        "software_id",
+        software_info_schema.SoftwareInfo,
+        is_test,
+    )
+
+    if not id_val:
+      print("software id validation failed")
+      return False
+    return True
```

Comment:
@stingram 
these should be utility functions provided as part of the benchmark writer.
can you fix it so this does not have to be replicated here?